src/init.cpp

```diff
@@ -937,36 +943,55 @@ bool AppInit2(Config &config, boost::thread_group &threadGroup, CScheduler &sche
     // ********************************************************* Step 6: load block chain
 
     fReindex = GetBoolArg("-reindex", DEFAULT_REINDEX);
+    int64_t requested_block_mode = GetArg("-blockdbtype", 1);
```

Comment:
use `DEFAULT_BLOCK_DB_MODE` as second argument of `GetArg` so you wouldn't need to bother to special case for the default value (next if)